src/SharpBrick.PoweredUp/Protocol/Formatter/PortValueSingleEncoder.cs

```diff
@@ -78,10 +78,10 @@ internal static PortValueData<sbyte> CreatePortValueDataSByte(PortModeInfo modeI
         {
             var rawValues = MemoryMarshal.Cast<byte, sbyte>(dataSlice).ToArray();
 
-            var siValues = rawValues.Select(rv => Scale(rv, modeInfo.RawMin, modeInfo.RawMax, modeInfo.SIMin, modeInfo.SIMax)).Select(f => Convert.ToSByte(f)).ToArray();
+            var siValues = rawValues.Select(rv => Scale(rv, modeInfo.RawMin, modeInfo.RawMax, modeInfo.SIMin, modeInfo.SIMax)).Select(f => ConvertSByteValueFromFloat(f)).ToArray();
             var pctValues = modeInfo.DisablePercentage switch
```

Comment:
What I did for overflowing percentages in the past is to deactive the percentage calculation ;)

We can leave this code in (it essentially prevents an exception) but I will revert it when the proplem is universally solved with #126.

Reply:
Maybe disabling percentage is easier then.  I'll revert the encoder changes and disable percentage and let it be resolved properly by #126.